projects/packages/sync/src/sync-queue/class-queue-storage-table.php

```diff
@@ -0,0 +1,471 @@
+		require_once ABSPATH . '/wp-admin/includes/upgrade.php';
+
+		$charset_collate = $wpdb->get_charset_collate();
+
+		$table_definition = "CREATE TABLE {$this->table_name} (
+			  `ID` bigint(20) NOT NULL AUTO_INCREMENT,
+			  `queue_id` varchar(50) NOT NULL,
+			  `event_id` varchar(100) NOT NULL,
+			  `event_payload` text NOT NULL,
+			  `timestamp` timestamp NOT NULL DEFAULT current_timestamp(),
+			  `object_type` varchar(100) DEFAULT NULL,
+			  `parent_object_id` bigint(20) DEFAULT NULL,
+			  PRIMARY KEY (`ID`),
+			  KEY `event_id` (`event_id`),
+			  KEY `queue_id` (`queue_id`),
+			  KEY `queue_id_event_id` (queue_id, event_id),
+			  KEY `timestamp` (`timestamp`),
+			  KEY `object_type` (`object_type`),
+			  KEY `parent_object_id` (`parent_object_id`)
+			) $charset_collate;";
+
+		/**
+		 * The function dbDelta will only return the differences. If the table exists, the result will be empty,
+		 * so let's run a check afterward to see if the table exists and is healthy.
+		 */
+		\dbDelta( $table_definition );
+	}
+
+	/**
+	 * Check if the Dedicated table actually exists.
+	 *
+	 * @return bool
+	 */
+	public function dedicated_table_exists() {
+		global $wpdb;
+
+		// Check if the table exists
+		$result = $wpdb->get_row(
+			$wpdb->prepare( 'SHOW TABLES LIKE %s', $this->table_name ),
+			ARRAY_N
+		);
+
+		if ( empty( $result ) || count( $result ) !== 1 || $result[0] !== $this->table_name ) {
+			return false;
+		}
+
+		return true;
+	}
+	/**
+	 * Check if the table is healthy, and we can read and write from/to it.
+	 *
+	 * @return bool If the dedicated table is available, and we can read and write from/to it.
+	 */
+	public function is_dedicated_table_healthy() {
+		global $wpdb;
+
+		if ( ! $this->dedicated_table_exists() ) {
+			return false;
+		}
+
+		// Try to read from the table
+
+		// Ignore the interpolated table name
+		// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$query = $wpdb->query( "SELECT count(`ID`) FROM {$this->table_name}" );
+
+		if ( ! $query ) {
+			// The query failed to select anything from the table, so there must be an issue reading from it.
+			return false;
+		}
+
+		if ( $wpdb->last_error ) {
+			// There was an error reading, that's not necessarily failing the query.
+			// TODO check if we need this error check.
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Disable usage of the dedicated table.
+	 *
+	 * @return void
+	 */
+	public function disable_dedicated_table_usage() {
+		// TODO disable the option to use the table
+		// TODO check if healthy
+		// TODO migrate to the main queue class.
+	}
+
+	/**
+	 * Drop the dedicated table as part of cleanup.
+	 *
+	 * @return bool If the table is cleared. It's using `is_dedicated_table_healthy` to check.
+	 */
+	public function drop_table() {
+		global $wpdb;
+
+		// TODO do we need to check if the table is healthy or not before dropping it?
```

Comment:
I'd suggest keeping it simple here, by only returning `$wpdb->query( "DROP TABLE {$this->table_name}" );`